app/controllers/invites_controller.rb

```diff
@@ -22,15 +22,17 @@ def show
     invite = Invite.find_by(invite_key: params[:id])
     if invite.present? && invite.redeemable?
       if current_user
-        InvitedUser.transaction do
-          invited_user = InvitedUser.find_or_initialize_by(user: current_user, invite: invite)
-          if invited_user.new_record?
-            invited_user.save!
-            Invite.increment_counter(:redemption_count, invite.id)
-            invite.invited_by.notifications.create!(
-              notification_type: Notification.types[:invitee_accepted],
-              data: { display_username: current_user.username }.to_json
-            )
+        if current_user != invite.invited_by
```

Comment:
Tiny, but I would merge the two conditions on the same line. Avoid avoid too many nested `if`s.

```rails
if current_user && current_user != invite.invited_by
```

Reply:
That would result in a different behavior, where the user would see a "this invitation is intended for new users" error message. I do not think that is desired.

Reply:
Not sure I understand.

I'm suggesting going from

```rails
  if current_user
     if current_user != invite.invited_by
        # code
     end
  end
```

to

```rails
  if current_user && current_user != invite.invited_by
    # code
  end
```

To remove one nesting.

How is it changing the behavior?